src/plugins/Miner.cpp

```diff
@@ -98,19 +171,11 @@ void Miner::OnUnitCreated(const sc2::Unit* unit_, Builder*) {
 }
 
 void Miner::OnUnitIdle(const sc2::Unit* unit_, Builder*) {
-    auto units = gAPI->observer().GetUnits(sc2::IsVisibleMineralPatch(),
-        sc2::Unit::Alliance::Neutral);
-
     switch (unit_->unit_type.ToType()) {
         case sc2::UNIT_TYPEID::PROTOSS_PROBE:
         case sc2::UNIT_TYPEID::TERRAN_SCV:
         case sc2::UNIT_TYPEID::ZERG_DRONE: {
```

Comment:
Now you can get rid of the switch and use `if (unit_->unit_type == gHub->GetCurrentWorkerType())` here, too.

Reply:
Fixed.